neural_compressor/torch/algorithms/weight_only/autoround.py

```diff
@@ -12,27 +12,156 @@
+                                            1 is input channel. Defaults to 1.
+        device (str, optional): choose device for compression. Defaults to cpu.
+        use_optimum_format (bool, optional): use the popular huggingface compression format.
+            1: compression_dim: weight = 1, zeros = 0 and both are transposed.
+            2: zeros -= 1 before compression. Why we need it?
+            3: g_idx: use same number for one group instead of recording the channel order.
+            4. parameter name changed, such as 'packed_weight' -> 'qweight'.
+            5. zeros is always needed even for sym.
+        inplace (bool, optional): Compress the model in place, or copy the model and compress it.
+
+    xpu args:
+        compression_dtype=torch.int8,
+        compression_dim=0,
+        use_optimum_format=False,
+        scale_dtype=convert_dtype_str2torch(config.scale_dtype),
+        device="xpu",
+    """
+    if inplace:
+        compressed_model = model
+    else:
+        compressed_model = copy.deepcopy(model)
+    if isinstance(weight_config, str):
+        with open(weight_config, "r") as f:
+            q_config = json.load(f)
+    else:
+        q_config = weight_config
+    for k, v in q_config.items():
+        logger.info(f"Packing {k}")
+        if "float" in v["data_type"]:
+            continue
+        dtype = v["data_type"]
+        num_bits = v["bits"]
+        group_size = v["group_size"]
+        sym = v["sym"]
+        scale_dtype = v["scale_dtype"]
+        m = get_module(compressed_model, k)
+        fp_weight = m.weight.data
+        scale, zp = v["scale"], v["zp"]
+        convert_dtype = scale_dtype
+        if not isinstance(scale, torch.Tensor):
+            scale = torch.tensor(scale, dtype=convert_dtype)
+            zp = torch.tensor(zp, dtype=torch.int32)
+            if device == "xpu":
+                scale = torch.tensor(v["scale"], dtype=torch.float32)
+                zp = None if sym else torch.tensor(v["zp"], dtype=torch.int32)
+        else:
+            if not inplace:
+                scale = scale.clone()
+                zp = zp.clone()
+            if device == "xpu":
+                # Please note that for XPU, the scale data type is forcibly set to fp32
+                scale = scale.to(dtype=torch.float32)
+                zp = None if sym else zp.to(dtype=torch.int32)
+            else:
+                scale = scale.to(dtype=convert_dtype)
+                zp = zp.to(dtype=torch.int32)
+        if is_transformers_imported() and isinstance(m, transformers.Conv1D):
+            fp_weight = fp_weight.t_().contiguous()
+        int_weight = quant_weight_w_scale(fp_weight, scale, zp, group_size, fp_weight.device)
+        int_weight = int_weight.type(torch.int32)
+        if isinstance(m, torch.nn.Linear):
+            in_features = m.in_features
+            out_features = m.out_features
+        elif is_transformers_imported() and isinstance(m, transformers.Conv1D):
```

Comment:
Oh, I see the difference, can we swap the `transformers.Conv1D` with `torch.nn.Linear`, so that we can reuse `pack_model`?

Reply:
now raised pr to AutoRound. https://github.com/intel/auto-round/pull/118